src/didactUri.ts

```diff
@@ -0,0 +1,106 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+'use strict';
+
+import * as url from 'url';
+import {getValue} from './utils';
+import * as vscode from 'vscode';
+
+export class DidactUri {
+	private commandId: string | undefined = undefined;
+	private completionMessage : string | undefined = undefined;
+	private errorMessage : string | undefined = undefined;
+	private projectFilePath : string | undefined = undefined;
+	private srcFilePath : string | undefined = undefined;
+	private extFilePath : string | undefined = undefined;
+	private text : string | undefined = undefined;
+	private user : string | undefined = undefined;
+
+	private context: vscode.ExtensionContext;
+
+	public constructor(incomingUrl: string, ctx : vscode.ExtensionContext ) {
+		this.context = ctx;
+		this.parseDidactUrl(incomingUrl);
+	}
+
+	public getCommandId(): string | undefined {
+		return this.commandId;
+	}
+
+	public getCompletionMessage() : string | undefined {
+		return this.completionMessage;
+	}
+
+	public getErrorMessage() : string | undefined {
+		return this.errorMessage;
+	}
+
+	public getProjectFilePath() : string | undefined {
+		return this.projectFilePath;
+	}
+
+	public getSrcFilePath() : string | undefined {
+		return this.srcFilePath;
+	}
+
+	public getExtFilePath() : string | undefined {
+		return this.extFilePath;
+	}
+
+	public getText() : string | undefined {
+		return this.text;
+	}
+
+	public getUser() : string | undefined {
+		return this.user;
+	}
+
+	private parseDidactUrl(incoming: string) {
```

Comment:
how about adding a return type definition?

Reply:
is it necessary? this is a private method to populate the object from what comes into the constructor

Reply:
no, its not but I mean why would we use typescript instead of javascript if we skip defining types? but thats just my opinion.

Reply:
I'm very confused. All of the "get" methods have return types. The parseDidactUrl method is a) private and b) simply populating all of the internal variables. What do you think it should return? It's not meant to be used externally.

Reply:
I think he means to return void: https://www.tutorialsteacher.com/typescript/typescript-void

Reply:
yeah, sorry for being not clear enough. I mean the point of typescript is to have typing, right? so why shouldn't we use it all the time? Actually I think we shouldn't handle private and public differently. But that is just my opinion. Treat it as optional.